src/xpcc/driver/radio/dw1000/deca_regs.hpp

```diff
@@ -0,0 +1,1341 @@
+/*<------------------------------------------------------------------------------------------------------------------
+ * @file    deca_regs.hpp
+ * @brief   DW1000 Register Definitions
+ *          This file supports C++ development for DW1000 enabled devices
+ *
+ * @attention
+ *
+ * Copyright 2013 =(c); Decawave Ltd, Dublin, Ireland.
+ * c++ Conversion: Marten Junga (2018)
+ *
+ * All rights reserved.
+ *
+ */
+
+#ifndef _DECA_REGS_H_
+#define _DECA_REGS_H_
+
+#include <xpcc/architecture/platform.hpp>
+
+
+/****************************************************************************//**
+ * @brief Bit definitions for register DEV_ID
+**/
+static constexpr uint32_t DEV_ID_ID               =0x00;            /*<Device ID register, includes revision info =(=0xDECA0130); */
+static constexpr uint32_t DEV_ID_LEN              =(4);
+/*<mask and shift */
+static constexpr uint32_t DEV_ID_REV_MASK         =0x0000000FUL;    /*<Revision */
+static constexpr uint32_t DEV_ID_VER_MASK         =0x000000F0UL;    /*<Version */
+static constexpr uint32_t DEV_ID_MODEL_MASK       =0x0000FF00UL;    /*<The MODEL identifies the device. The DW1000 is device type =0x01 */
+static constexpr uint32_t DEV_ID_RIDTAG_MASK      =0xFFFF0000UL;    /*<Register Identification Tag =0xDECA */
```

Comment:
Was the original file done with a lot of `#define`s? And if so why did you translate this to C++ constexpr?

Reply:
Historical reasons my predecessor did that. I don't know why

Reply:
Ok, then this is original work, since you cannot copyright the data, only the formatting. Can you put your predecessor and yourself as copyright owners (BSD-licensed)?

I would then not put this into `ext/`, since we also don't put the register definitions of our sensor devices from `modm/driver` into `ext/` either.